Net/SMTP.php

```diff
@@ -749,6 +753,14 @@ public function auth($uid, $pwd , $method = '', $tls = true, $authz = '')
                 /* Return the PEAR_Error object from _getBestAuthMethod(). */
                 return $method;
             }
+        } elseif ($method === 'OAUTH') {
+            // special case of OAUTH, use the supported method
+            $available_methods = explode(' ', $this->esmtp['AUTH']);
+            foreach (['OAUTHBEARER', 'XOAUTH2'] as $method) {
+                if (in_array($method, $available_methods)) {
+                    break;
+                }
+            }
```

Comment:
Here you should raiseError (as below) if both aren't supported.

Reply:
you are right, the next raiseError may create a confusion, will add an explicit message